tests/base/071.phpt

```diff
@@ -0,0 +1,51 @@
+--TEST--
+parallel recursion in arrays in the closure
+--SKIPIF--
+<?php
+if (!extension_loaded('parallel')) {
+  die("skip parallel not loaded");
+}
+if (!version_compare(PHP_VERSION, "8.1", ">=")) {
+    die("skip php 8.1 required");
+}
+?>
+--FILE--
+<?php
+function transformChunk($n)
+{
+  $fibonacci = function ($n) use (&$fibonacci) {
+    if ($n == 0) {
+      return 0;
+    }
+    if ($n == 1) {
+      return 1;
+    }
+    return $fibonacci($n - 1) + $fibonacci($n - 2);
+  };
+  return $fibonacci($n);
+}
+
+$runtime = new \parallel\Runtime();
+$future = $runtime->run(
+  transformChunk(...),
+  [
+    10
+  ]
+);
+var_dump($future->value());
+
+$runtime = new \parallel\Runtime();
+$future = $runtime->run(
+  transformChunk(...),
+  [
+    10
+  ]
+);
+var_dump($future->value());
+
+?>
+--EXPECT--
+int(55)
+int(55)
+--XLEAK--
+REASON: no cyclic reference collector implemented yet
```

Comment:
I would expect all memory to he related no latter than the 1st GC run.

Reply:
That'd be true "normally", but the memory leaking is not ref counted 😉 
The leak is copies I have to create for the closure passed into `\parallel\run()` to execute it in another thread. So this leak is not referring to user land memory, but extension allocated memory.

Reply:
@realFlowControl do I understand this correctly that any memory used in a thread isn't cleaned up until the thread dies? (Hoping I misunderstood it.)

Reply:
The leak this comment is referring to is not userland memory (whatever you allocate in PHP). It is referring to a few bytes that parallel is allocating with the system allocator and it was there since - most likely - ever. So this just makes this explicit and I'll need to set aside some time to solve this issue. In this PR I've also made this explicit for `tests/base/068.phpt` and `tests/base/069.phpt` which where failing for exactly that reason also, now it just clear that they are not "failing", but failing because of leaking memory.

Reply:
@WyriHaximus adding to this: parallel only leaks if you pass a variable that has a cyclomatic reference like this thing:
```php
$array = [];
$array["self"] = &$array;

\parallel\run(function($array){
    var_dump($array);
}, [$array]);
```
If you don't, nothing leaks ... so far (until this PR) this would crash with a segfault, now it works but leaks a few bytes. Next up is tracking down the memory leak ...

Reply:
If there is a memory leak due circular reference, I think the issue is the reference is hold by this extension because the extension does not participate in GC cycle, and thus the refcount never reaches 0 when running GC.